File: src/js/bruker2dicom/modules/study.js
require('bruker2dicom/common.js');

var _module = namespace('dicomifier.bruker2dicom.modules');

_module.GeneralStudy = function(indexGenerator, dicomDataset, brukerDataset) {
    var dateTimeMapper = dicomifier.bruker2dicom.dateTimeMapper;
    var toDicom = dicomifier.bruker2dicom.toDicom;

    toDicom(
        indexGenerator, dicomDataset, 'StudyInstanceUID',
        brukerDataset, 'VisuStudyUid', 1);

    toDicom(
        indexGenerator, dicomDataset, 'StudyDate', brukerDataset,
        'VisuStudyDate', 2, dateTimeMapper('date'));

    toDicom(
        indexGenerator, dicomDataset, 'StudyTime', brukerDataset,
        'VisuStudyDate', 2, dateTimeMapper('time'));

    toDicom(
        indexGenerator, dicomDataset, 'ReferringPhysicianName',
        brukerDataset, 'VisuStudyReferringPhysician', 2);

    toDicom(
        indexGenerator, dicomDataset, 'StudyID',
        brukerDataset, 'VisuStudyNumber', 2);

    dicomDataset[dicomifier.dictionary['AccessionNumber'][1]] = {
        'vr': dicomifier.dictionary['AccessionNumber'][0] };

    toDicom(
        indexGenerator, dicomDataset, 'StudyDescription',
        brukerDataset, 'VisuStudyId', 3);
};

_module.PatientStudy = function(indexGenerator, dicomDataset, brukerDataset) {
    var dateTimeMapper = dicomifier.bruker2dicom.dateTimeMapper;
    var toDicom = dicomifier.bruker2dicom.toDicom;

    toDicom(
        indexGenerator, dicomDataset, 'PatientWeight',
        brukerDataset, 'VisuSubjectWeight', 3, parseFloat);

    dicomDataset[dicomifier.dictionary['PatientSexNeutered'][1]] = {
        'vr': dicomifier.dictionary['PatientSexNeutered'][0] };
};

File: src/python/dicomifier/bruker_to_dicom/modules/study.py
#########################################################################
# Dicomifier - Copyright (C) Universite de Strasbourg
# Distributed under the terms of the CeCILL-B license, as published by
# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
# for details.
#########################################################################

GeneralStudy = [ # PS 3.3, C.7.2.1
    ("VisuStudyUid", "StudyInstanceUID", 1, None, None),
    ("VisuStudyDate", "StudyDate", 2, None, None),
    ("VisuStudyDate", "StudyTime", 2, None, None),
    ("VisuStudyReferringPhysician", "ReferringPhysicianName", 2, None, None),
    ("VisuStudyNumber", "StudyID", 2, None, lambda v: [str(x) for x in v]),
    (None, "AccessionNumber", 2, lambda d,g,i: None, None),
    ("VisuStudyId", "StudyDescription", 3, None, None),
]

PatientStudy = [ # PS 3.3, C.7.2.2.html
    ("VisuSubjectWeight", "PatientWeight", 3, None, None),
    (None, "PatientSexNeutered", 2, lambda d,g,i: None, None),
]
